### PR TITLE
Provision environment variables for initContainers of chePlugin/cheEditor

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Provisions environment variables for initContainers of chePlugin/cheEditor.

The only init container we have now for chePlugin/cheEditor - remote-plugin-laucher and I'm not sure if it makes sense to tune it. But I think if we provision additional, unused env vars to init containers - it won't hurt anyone in most cases but potentially can be used if there is need to tune initContainer of chePlugin/cheEditor and we will have consistent behavior for all component types.

### What issues does this PR fix or reference?
If follows up https://github.com/eclipse/che/pull/15435

### How to test this PR 
Che Server image build by Happy Path job: maxura/che-server:15508
<details>

<summary>Devfile</summary>

```yaml
metadata:
  name: env-override-new
components:
  - type: kubernetes
    reference: >-
      https://gist.githubusercontent.com/sleshchenko/86d28a2c4426bdd0cef57bdfe3c7a829/raw/555b20c10ff7d6519b80d0841a30735a2f4f2118/deployment.yaml
    alias: env-test
    env:
      - value: Tom
        name: NAME
  - memoryLimit: 1G
    type: cheEditor
    reference: >-
      https://gist.githubusercontent.com/sleshchenko/5e1118c93a17e6b7d4efc4cd6f88b164/raw/b90a076ab92e3a14101f0275e0573b3ce2eff49d/meta.yaml
    alias: editor
    env:
      - value: Tom
        name: NAME
  - id: ms-vscode/go/latest
    memoryLimit: 256M
    type: chePlugin
    alias: plugin
    env:
      - value: Tom
        name: NAME
apiVersion: 1.0.0
commands:
  - name: test-initContainers
    actions:
      - type: exec
        command: cd /hello && ls
        component: env-test
  - name: test-editor
    actions:
      - type: exec
        command: 'echo Hello ${NAME}'
        component: editor
  - name: test-plugin
    actions:
      - type: exec
        command: 'echo Hello ${NAME}'
        component: plugin
```

</details>

#### Release Notes
N/A

#### Docs PR
A note will be added to che docs if reviewers think it makes sense to provision env vars to initContainers as well.

### CODE REVIEW

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/MachineResolver.java [hunk 1]
```diff
@@ -67,7 +66,7 @@ public InternalMachineConfig resolve() throws InfrastructureException {
     InternalMachineConfig machineConfig =
         new InternalMachineConfig(
             toServers(containerEndpoints),
-            component.getEnv().stream().collect(Collectors.toMap(Env::getName, Env::getValue)),
+            emptyMap(),
```
Comment (amisevsk) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/MachineResolver.java:69]: I don't think I have the full picture, but I don't understand why this change is required.

**Review by amisevsk — COMMENTED** [review]
Changes LGTM, though using the posted devfile I encounter an error starting the workspace:

`Unable to start workspace. Error when trying to start the workspace: Failure executing: POST at: https://172.30.0.1/apis/apps/v1/namespaces/amisevsk-che-2/deployments. Message: Deployment.apps "workspace0pe2qmn7e8843zap.empty-env" is invalid: spec.template.spec.volumes[2].name: Duplicate value: "hello". Received status: Status(apiVersion=v1, code=422, details=StatusDetails(causes=[StatusCause(field=spec.template.spec.volumes[2].name, message=Duplicate value: "hello", reason=FieldValueDuplicate, additionalProperties={})], group=apps, kind=Deployment, name=workspace0pe2qmn7e8843zap.empty-env, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=Deployment.apps "workspace0pe2qmn7e8843zap.empty-env" is invalid: spec.template.spec.volumes[2].name: Duplicate value: "hello", metadata=ListMeta(_continue=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=Invalid, status=Failure, additionalProperties={}).
`

I don't see what would cause this though.

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/MachineResolver.java (hunk 1 above)
Comment (sleshchenko) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/MachineResolver.java:69]: Now we propagate environment variable directly to K8s Containers objects instead of reusing deprecated `MachineConfig` that will be applied to K8s objects on Provisioning phase.

**Comment by sleshchenko** [comment]
```
Unable to start workspace. ... Deployment.apps "workspace0pe2qmn7e8843zap.empty-env" is invalid: spec.template.spec.volumes[2].name: Duplicate value: "hello". ...
```
@amisevsk Thanks for catching it. On my Che installation, I don't face it, I think Che has some issue in volumes provisioning and depending on PVC strategy it can be reproduced. In my case a common PVC strategy is used, which PVC strategy do you use?

**Comment by amisevsk** [comment]
@sleshchenko My deployment uses the `unique` strategy

**Comment by sleshchenko** [comment]
Seems the same issue happens on master branch as well https://github.com/eclipse/che/issues/15576

**Review by skabashnyuk — APPROVED** [review]

**Comment by amisevsk** [comment]
https://github.com/eclipse/che/issues/15576 is a slightly different issue AFAIK, where an earlier version of codeserver had `mountSources: true` and explicitly included a mount for `/projects`

**Comment by sleshchenko** [comment]
@amisevsk I've reproduced the same issue you faced on the nightly image, so, it's not something that this PR introduces. I'll create a separate issue to handle it
Screenshot_20200204_100827

**Review by amisevsk — APPROVED** [review]
LGTM if the issue is in master.
